### ISSUE TITLE
f/sessionHandler destroy

### ISSUE THREAD

**Issue by josemartinez** [canni/YiiMongoDbSuite#35]
Hi Canni, 
I have tested your new feature sessionHandler in my application and i'm having a trouble with it. I dont know if its a bug on the code or not. 

Step1
I connect to the login page of my application
A record is created on database with an Id, expire and empty data

Step2
I login on the application
Record is updated on database with some data='xxxxx'

Step3 
I logout using Yii::app()->user->logout(); (i didnt extend user, so im using CWebUser) 
Logout seems to be successful, but i still have data='xxxx', so application doesnt knows that user logged out.

After some debug i noticed that we call once to EMongoHttpSession->destroySession and i see that we remove correctly the record with the id, but later it seems that 'data' field is restored from .... maybe SESSION? (asynchronous treatement between database and session, i dont know..) 

I wanted to know if you noticed this behavior or i just have something wrong in my application. 

Thank you very much

Best regards

**Comment by canni** [comment]
Hey, thanks for info, I've written the mongo session handler out of hand, and I do not tested it, so I suppose that there is a bug in my implementation, will try to figure it out.

**Comment by dhastings** [comment]
I ran into this issue as well and found that changing (in the sessionDestroy method):

`$this->_collection->remove(array('_id'=>$id, array('justOne'=>true)));`

to

``` $this->_collection->remove(array('_id'=>$id));```

Resolved the issue. I think it's safe to not use the justOne flag. The _id is a mongo object id and as such it there can't be two in the collection anyway.

Other than that, EMongoHttpSession works fine from our testing here at the office. 
```
